app/services/clients/publishing_api.rb

```diff
@@ -13,33 +13,42 @@ def initialize
       @per_page = 100
     end
 
-    def find_each(fields)
+    def find_each(fields, options = {})
       current_page = 1
+
+      query = build_query(fields, options)
+
       loop do
-        response = publishing_api.get_content_items(
-          document_type: 'taxon',
-          order: '-public_updated_at',
-          q: '',
-          page: current_page,
-          per_page: per_page,
-          states: ['published'],
-        )
-
-        taxonomies = map_taxon_results(response['results'], fields)
-        taxonomies.each { |taxon| yield taxon }
+        query = page_query(query, current_page)
+        response = publishing_api.get_content_items(query)
+        response["results"].each { |result| yield result.symbolize_keys }
+
         break if last_page?(response)
         current_page = response["current_page"] + 1
       end
     end
 
   private
 
-    def last_page?(response)
-      response["pages"] == response["current_page"]
+    def build_query(fields, options)
+      {
+        document_type: options[:document_type] || 'taxon',
```

Comment:
I would remove the default; it might be better to fail than to return unexpected information. What do you think?